app/assets/javascripts/discourse/components/search-advanced-options.js.es6

```diff
@@ -74,107 +86,204 @@ export default Em.Component.extend({
   },
 
   _update() {
-    let searchTerm = this.get('searchTerm');
-
-    if (!searchTerm) {
+    if (!this.get('searchTerm')) {
       this._init();
       return;
     }
 
-    this.findUsername(searchTerm);
-    this.findCategory(searchTerm);
-    this.findGroup(searchTerm);
-    this.findBadge(searchTerm);
-    this.findTags(searchTerm);
-    this.findIn(searchTerm);
-    this.findStatus(searchTerm);
-    this.findPostsCount(searchTerm);
-    this.findPostTime(searchTerm);
+    this.findSearchTerms();
+    this.setSearchedTermValue('searchedTerms.username', REGEXP_USERNAME_PREFIX);
+    this.setSearchedTermValueForCategory();
+    this.setSearchedTermValueForGroup();
+    this.setSearchedTermValueForBadge();
+    this.setSearchedTermValueForTags();
+    this.setSearchedTermValue('searchedTerms.in', REGEXP_IN_PREFIX, REGEXP_IN_MATCH);
+    this.setSearchedTermSpecialInValue('searchedTerms.special.in.likes', REGEXP_SPECIAL_IN_LIKES_MATCH);
+    this.setSearchedTermSpecialInValue('searchedTerms.special.in.private', REGEXP_SPECIAL_IN_PRIVATE_MATCH);
+    this.setSearchedTermSpecialInValue('searchedTerms.special.in.wiki', REGEXP_SPECIAL_IN_WIKI_MATCH);
+    this.setSearchedTermValue('searchedTerms.status', REGEXP_STATUS_PREFIX);
+    this.setSearchedTermValueForPostTime();
+    this.setSearchedTermValue('searchedTerms.posts_count', REGEXP_POST_COUNT_PREFIX);
   },
 
-  findSearchTerm(EXPRESSION, searchTerm) {
+  findSearchTerms() {
+    const searchTerm = this.get('searchTerm');
     if (!searchTerm)
-      return "";
+      return [];
 
-    const expressionPosition = searchTerm.search(EXPRESSION);
-    if (expressionPosition === -1)
-      return "";
+    let result = [];
+    const blocks = searchTerm.match(REGEXP_BLOCKS);
+    _.each(blocks, function(block) {
```

Comment:
Okay, but we still have the dependency due to _.isArray checks also in this file.